lawnchair/src/app/lawnchair/ui/preferences/components/AskForFilePermission.kt

```diff
@@ -0,0 +1,59 @@
+package app.lawnchair.ui.preferences.components
+
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.requiredWidth
+import androidx.compose.material3.Button
+import androidx.compose.material3.OutlinedButton
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import app.lawnchair.preferences.PreferenceAdapter
+import app.lawnchair.preferences.PreferenceManager
+import app.lawnchair.ui.ModalBottomSheetContent
+import app.lawnchair.ui.preferences.components.controls.SwitchPreference
+import app.lawnchair.util.checkAndRequestFilesPermission
+import com.android.launcher3.R
+
+@Composable
+fun AskForFileAccessPermission(
+    adapter: PreferenceAdapter<Boolean>,
+    modifier: Modifier = Modifier,
+    onDismissRequest: () -> Unit,
+) {
+    val context = LocalContext.current
+
+    ModalBottomSheetContent(
+        modifier = modifier,
+        title = {
+            Text(text = "File access required")
```

Comment:
[GENERAL] Avoid using literal string to ensures translatability

(192.168.1.37 • @CollisionBot CI)